src/test/java/net/openhft/chronicle/queue/impl/single/NormaliseEOFsTest.java

```diff
@@ -0,0 +1,96 @@
+package net.openhft.chronicle.queue.impl.single;
+
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.core.io.IOTools;
+import net.openhft.chronicle.core.onoes.ExceptionKey;
+import net.openhft.chronicle.core.onoes.LogLevel;
+import net.openhft.chronicle.core.time.SetTimeProvider;
+import net.openhft.chronicle.core.time.TimeProvider;
+import net.openhft.chronicle.queue.ExcerptAppender;
+import net.openhft.chronicle.queue.QueueTestCommon;
+import net.openhft.chronicle.queue.rollcycles.TestRollCycles;
+import net.openhft.chronicle.testframework.internal.ExceptionTracker;
+import net.openhft.chronicle.wire.DocumentContext;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class NormaliseEOFsTest extends QueueTestCommon {
+
+    private static final String LOG_LEVEL_PROPERTY = "org.slf4j.simpleLogger.log." + StoreAppender.class.getName();
+    private static final String QUEUE_PATH = "normaliseEOFsTest";
+    private Map<ExceptionKey, Integer> exceptionMap;
+
+    @Before
+    public void setLogLevelProperty() {
+        System.setProperty(LOG_LEVEL_PROPERTY, "debug");
+    }
+
+    @Before
+    public void clearDataFromPreviousRun() {
+        IOTools.deleteDirWithFilesOrThrow(QUEUE_PATH);
+    }
+
+    @Before
+    @Override
+    public void recordExceptions() {
+        super.recordExceptions();
+        exceptionMap = Jvm.recordExceptions(true);
+        exceptionTracker = ExceptionTracker.create(ExceptionKey::message, ExceptionKey::throwable, Jvm::resetExceptionHandlers, exceptionMap,
+                (key) -> key.level != LogLevel.DEBUG && key.level != LogLevel.PERF,
+                (key) -> key.level() + " " + key.clazz().getSimpleName() + " " + key.message());
+        ignoreException(ex -> true, "Ignore everything");
```

Comment:
This is a bit naughty. We probably should add a way to ask for debug exceptions and iterate over the exceptions in order.